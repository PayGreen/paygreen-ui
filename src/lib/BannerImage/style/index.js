import styled from 'styled-components';
import { decorationOptions } from '../../../shared/constants';
import { topStyle, bottomStyle } from './base';

const BannerImageBase = styled.div`
    ${props => topStyle[props.topStyle]};
    ${props => bottomStyle[props.bottomStyle]};
    position: relative;
    min-height: ${({theme}) => theme.bannerHeight.sm};
    padding: 0 .1px ${({theme}) => theme.blockSpace.md} .1px;
    padding-top: ${({topStyle, theme}) => topStyle !== decorationOptions.none ? theme.wave.simple : 0};
    padding-bottom: ${({bottomStyle, theme}) => bottomStyle !== decorationOptions.none ? theme.wave.simple : 0};

    &::before,
    &::after {
        content: '';
        position: absolute;
        z-index: ${({theme}) => theme.zindex.background};
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        max-height: ${({theme}) => theme.bannerHeight.lg};
        mask-image: linear-gradient(to top, transparent, black ${({theme}) => theme.bannerHeight.sm});
    }

    &::before {
        background-image: url('${({image}) => image}');
        background-size: cover;
        filter: grayscale(1);
        opacity: .5;
    }

    &::after {
        background-color: ${({useColorTheme, theme, colorTheme}) => useColorTheme ? theme.color[colorTheme].light : null};
        opacity: .8;
    }
`;

export { BannerImageBase };