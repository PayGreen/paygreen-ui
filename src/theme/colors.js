const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
