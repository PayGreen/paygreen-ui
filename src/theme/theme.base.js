const responsive = {
    sm: 550,
    md: 960,
    lg: 1200,
    xl: 1360,
};

const query = (value, operator) => {
    if (operator == 'max') {
        value -= 1;
    }
    return operator + '-width: ' + value + 'px';
};

const pallet = {
    green: {
        main: '#33ad73',
        light: '#d5f3e5',
    },
    blue: {
        main: '#00a99d',
        light: '#e1f3f2',
    },
    orange: {
        main: '#e58a28',
        light: '#fbe6ce',
    },
};

export const ThemeBase = {
    name: 'Base',
    color: {
        primary: {
            ...pallet.green,
            gradientBase: '#77d1a6',
            gradientShade: '#a4db95',
        },
        secondary: {
            ...pallet.blue,
            gradientBase: '#51c6be',
            gradientShade: '#76bbd8',
        },
        tertiary: {
            ...pallet.orange,
            gradientBase: '#fab56b',
            gradientShade: '#faa272',
        },
        quaternary: {
            main: '#ec6a6a',
            light: '#ffe0e1',
            gradientBase: '#e98e90',
            gradientShade: '#e98eb9',
        },
        quinary: {
            main: '#a26ae1',
            light: '#e9d9fb',
            gradientBase: '#b887ee',
            gradientShade: '#dd94ef',
        },
    },
    status: {
        success: {
            ...pallet.green,
        },
        default: {
            ...pallet.blue,
        },
        warning: {
            ...pallet.orange,
        },
        danger: {
            main: '#b54025',
            light: '#f7d7d0',
        },
    },
    font: {
        size: {
            tiny: '1rem',
            xxs: '1.2rem',
            xs: '1.4rem',
            sm: '1.6rem',
            base: '1.8rem',
            md: '2.1rem',
            lg: '3rem',
            xl: '4rem',
        },
        weight: {
            normal: 400,
            bold: 700,
        },
        lineHeight: {
            base: 1,
            md: 1.5,
        },
        spacing: '0.3em',
        underline: {
            space: {
                tiny: '8px',
                xxs: '10px',
                xs: '10px',
                sm: '12px',
                base: '12px',
                md: '12px',
                lg: '24px',
                xl: '24px',
            },
            line: {
                tiny: '2px',
                xxs: '2px',
                xs: '2px',
                sm: '2px',
                base: '2px',
                md: '2px',
                lg: '3px',
                xl: '3px',
            },
            width: {
                tiny: '30px',
                xxs: '40px',
                xs: '40px',
                sm: '50px',
                base: '50px',
                md: '65px',
                lg: '80px',
                xl: '80px',
            },
        },
    },
    line: '2px',
    dash: '3px',
    radius: {
        none: '0',
        sm: '4px',
        md: '8px',
        lg: '12px',
    },
    arrow: {
        sm: '8px',
        md: '12px',
        lg: '16px',
    },
    space: {
        none: '0',
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '32px',
        xl: '64px',
    },
    grid: {
        header: '45px',
        headerBig: '70px',
        topbar: '45px',
        sidebar: '380px',
        maxWidth: '1400px',
    },
    blockShift: {
        none: '0',
        xs: '64px',
        sm: '128px',
        md: '256px',
        lg: '320px',
        xl: '384px',
    },
    blockSpace: {
        none: '0',
        xs: '16px',
        sm: '24px',
        md: '40px',
        lg: '52px',
        xl: '68px',
    },
    blockWidth: {
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '800px',
    },
    blockHeader: {
        sm: '140px',
        md: '180px',
        lg: '210px',
        xl: '240px',
    },
    bannerHeight: {
        xs: '33vh',
        sm: '50vh',
        lg: '100vh',
    },
    wave: {
        simple: '8vw',
        double: '11vw',
    },
    icon: {
        size: {
            xxs: '12px',
            xs: '16px',
            sm: '20px',
            md: '24px',
            lg: '32px',
            xl: '40px',
        },
        shift: {
            xxs: '2px',
            xs: '3px',
            sm: '4px',
            md: '4px',
            lg: '6px',
            xl: '6px',
        },
    },
    imageSize: {
        auto: 'auto',
        xxs: '150px',
        xs: '240px',
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '760px',
    },
    imageCircleSize: {
        auto: 'auto',
        xxs: '38px',
        xs: '60px',
        sm: '80px',
        md: '100px',
        lg: '120px',
        xl: '140px',
    },
    logoHeight: {
        none: 'auto',
        xs: '20px',
        sm: '30px',
        md: '40px',
        lg: '50px',
        xl: '60px',
    },
    logoWidth: {
        none: 'auto',
        xs: '90px',
        sm: '120px',
        md: '150px',
        lg: '200px',
        xl: '250px',
    },
    loaderSize: {
        xxs: '40px',
        xs: '50px',
        sm: '60px',
        md: '70px',
        lg: '80px',
        xl: '90px',
    },
    transition: {
        xxs: '0.1s',
        xs: '0.2s',
        sm: '0.3s',
        md: '0.5s',
        lg: '0.7s',
        xl: '1s',
    },
    zindex: {
        hidden: -100,
        layer: -5,
        background: -1,
        base: 0,
        content: 1,
        corner: 3,
        front: 5,
        loader: 90,
        subHeader: 100,
        menu: 105,
        header: 110,
        popin: 115,
        modal: 120,
    },
    query: {
        min: {
            sm: query(responsive.sm, 'min'),
            md: query(responsive.md, 'min'),
            lg: query(responsive.lg, 'min'),
            xl: query(responsive.xl, 'min'),
        },
        max: {
            sm: query(responsive.sm, 'max'),
            md: query(responsive.md, 'max'),
            lg: query(responsive.lg, 'max'),
            xl: query(responsive.xl, 'max'),
        },
    },
    button: {
        paddingWidth: {
            sm: '22px',
            md: '26px',
            lg: '30px',
        },
        paddingHeight: {
            sm: '12px',
            md: '14px',
            lg: '16px',
        },
        shift: '3px',
        font: {
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.5rem',
        },
        letterSpacing: {
            sm: '0.3rem',
            md: '0.4rem',
            lg: '0.45rem',
        },
    },
    daButton: {
        buttonHeight: {
            sm: '32px',
            md: '38px',
            lg: '44px',
        },
        shift: {
            sm: '2px',
            md: '4px',
            lg: '6px',
        },
        font: {
            sm: '1.35rem',
            md: '1.4rem',
            lg: '1.7rem',
        },
        iconSpacing: {
            sm: '0.9rem',
            md: '1rem',
            lg: '1.1rem',
        },
    },
    form: {
        inputWidth: {
            xs: '260px',
            sm: '330px',
            md: '400px',
            lg: '500px',
        },
        textareaHeight: '200px',
        radio: {
            md: '120px',
            lg: '150px',
        },
        checkboxSize: {
            sm: '18px',
            md: '20px',
            lg: '22px',
        },
        radioButtonSize: {
            sm: '19px',
            md: '21px',
            lg: '23px',
        },
        radioButtonSpace: {
            sm: '8px',
            md: '10px',
            lg: '10px',
        },
        toggle: '30px',
    },
    corner: {
        size: {
            lg: '34px',
        },
        padding: {
            lg: '6px',
        },
    },
    shadow: {
        size: {
            none: '0 0 0',
            xs: '1px 1px 5px',
            sm: '1px 1px 10px',
            md: '2px 2px 15px',
            lg: '3px 3px 20px',
        },
        opacity: {
            none: 1,
            xs: 0.95,
            sm: 0.9,
            md: 0.8,
            lg: 0.7,
        },
    },
    rotateSize: {
        d0: '0deg',
        d45: '45deg',
        d90: '90deg',
        d135: '135deg',
        d180: '180deg',
        d225: '225deg',
        d270: '270deg',
        d315: '315deg',
    },
    dataBar: {
        height: '10px',
        width: {
            sm: '300px',
            md: '400px',
            lg: '500px',
            xl: '600px',
        },
    },
};
