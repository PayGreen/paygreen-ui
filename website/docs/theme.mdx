---
id: theme
title: üñåÔ∏è Constantes du th√®me
sidebar_position: 2
---

Les composants de PayGreen UI utilisent tous de nombreuses constantes (couleurs, tailles, media queries&hellip;) d√©finies dans le th√®me. Certaines de ces constantes sont int√©ressantes √† conna√Ætre afin de mieux apr√©hender le fonctionnement de certains composants.

## Tailles

### Breakpoints des media queries

Ces constantes sont accessibles via `theme.breakpoint['la valeur']`. Pour obtenir les media queries correspondantes, il faut utiliser `theme.screen.min['la valeur']` (pour un √©cran d'une taille minimale √©gale √† la valeur renseign√©e) et `theme.screen.max['la valeur']` (pour un √©cran inf√©rieur √† la valeur renseign√©e).

| Valeur du `breakpoint` | Largeur d'√©cran |
| ---------------------- | --------------- |
| `sm`                   | 550px           |
| `md`                   | 960px           |
| `lg`                   | 1200px          |
| `xl`                   | 1360px          |

### Spaces

Ces constantes sont accessibles via `theme.space['la valeur']` et sont utilis√©es pour beaucoup de styles diff√©rents (paddings, largeurs&hellip;).

| Valeur du `space` | Taille |
| ----------------- | ------ |
| `none`            | 0      |
| `xs`              | 4px    |
| `sm`              | 8px    |
| `md`              | 16px   |
| `lg`              | 32px   |
| `xl`              | 64px   |

:::caution

Cette documentation est en cours de r√©daction et n'est pas encore compl√®te.

:::
