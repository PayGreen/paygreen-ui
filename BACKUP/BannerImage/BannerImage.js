import React from 'react';
import PropTypes from 'prop-types';
import {
    gradientOptions,
    colorThemeOptions,
    colorThemeDefault,
    decorationOptions,
    decorationDefault,
    decorationImageOptions,
    decorationImageDefault,
} from '../../shared/constants';
import { BannerImageBase } from './style';

const BannerImage = props => {
    return <BannerImageBase {...props}>{props.children}</BannerImageBase>;
};

BannerImage.propTypes = {
    image: PropTypes.string.isRequired,
    hasModifiedColor: PropTypes.bool,
    gradient: PropTypes.oneOf(Object.values(gradientOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    topStyle: PropTypes.oneOf(Object.values(decorationOptions)),
    bottomStyle: PropTypes.oneOf(Object.values(decorationImageOptions)),
};

BannerImage.defaultProps = {
    hasModifiedColor: true,
    gradient: gradientOptions.theme,
    colorTheme: colorThemeDefault,
    topStyle: decorationDefault,
    bottomStyle: decorationImageDefault,
};

export default BannerImage;
