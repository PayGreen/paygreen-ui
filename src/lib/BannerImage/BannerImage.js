import React from 'react';
import PropTypes from 'prop-types';
import {
    gradientOptions,
    colorThemeOptions,
    colorThemeDefault,
    decorationOptions,
    decorationDefault,
} from '../../shared/constants';
import { BannerImageBase } from './style';

const BannerImage = props => {
    return <BannerImageBase {...props}>{props.children}</BannerImageBase>;
};

BannerImage.propTypes = {
    gradient: PropTypes.oneOf(Object.values(gradientOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    topStyle: PropTypes.oneOf(Object.values(decorationOptions)),
};

BannerImage.defaultProps = {
    gradient: gradientOptions.theme,
    colorTheme: colorThemeDefault,
    topStyle: decorationDefault,
};

export default BannerImage;
