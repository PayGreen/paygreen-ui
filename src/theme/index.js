import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
