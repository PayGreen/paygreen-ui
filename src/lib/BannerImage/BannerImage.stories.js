import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios } from '@storybook/addon-knobs';
import {
    folder,
    colorThemeOptions,
    colorThemeDefault,
    decorationOptions,
    decorationDefault,
    greyOptions,
    fontSizeOptions,
    spaceOptions,
} from '../../shared/constants';
import Card from '../Card/Card';
import Title from '../Title/Title';
import Text from '../Text/Text';
import BannerImage from './BannerImage';
import photoFile from './sample/sample.png';

const colorThemeLabel = 'ColorTheme';

storiesOf(folder.block + 'BannerImage', module)
    .addDecorator(withKnobs)
    .add('BannerImage', () => (
        <BannerImage
            colorTheme={radios(
                colorThemeLabel,
                colorThemeOptions,
                colorThemeDefault,
            )}
            topStyle={radios(
                'Wave top direction',
                decorationOptions,
                decorationDefault,
            )}
            image={photoFile}
        >
            <Card hasBackground={false}>
                <Title
                    colorTheme={radios(
                        colorThemeLabel,
                        colorThemeOptions,
                        colorThemeDefault,
                    )}
                    colorWab={greyOptions.grey60}
                    textSize={fontSizeOptions.lg}
                    marginLateral={spaceOptions.md}
                    marginTop={spaceOptions.md}
                >
                    Your title with <strong>an emphasis</strong>
                </Title>

                <Text
                    marginLateral={spaceOptions.md}
                    marginTop={spaceOptions.md}
                >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Duis porttitor velit a ultricies aliquet. Donec vehicula in
                    arcu non sodales.
                </Text>
            </Card>
        </BannerImage>
    ));
