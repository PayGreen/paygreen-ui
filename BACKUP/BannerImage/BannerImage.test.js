import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import BannerImage from './BannerImage';
import photoFile from './sample/sample.png';

it('renders without crashing', () => {
    const bannerImage = TestRenderer.create(
        <BannerImage theme={ThemeDefault} image={photoFile}>
            <div>
                Something
            </div>
        </BannerImage>
    );
    expect(bannerImage.toJSON()).toMatchSnapshot();
});