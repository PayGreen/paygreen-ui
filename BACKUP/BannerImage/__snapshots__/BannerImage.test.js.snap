// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__BannerImageBase-sc-47mmou-0 jzlEeT"
>
  <div>
    Something
  </div>
</div>
`;
